internal/auth0/role/resource_permissions_test.go

```diff
@@ -0,0 +1,105 @@
+		name = "read:foo"
+	}
+}
+
+data "auth0_role" "role" {
+	depends_on = [ auth0_role_permissions.role_permissions ]
+	role_id = auth0_role.role.id
+}
+`
+
+func TestAccRolePermissions(t *testing.T) {
+	acctest.Test(t, resource.TestCase{
+		Steps: []resource.TestStep{
+			{
+				Config: acctest.ParseTestName(testAccRolePermissionNoneAssigned, strings.ToLower(t.Name())),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.auth0_role.role", "permissions.#", "0"),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(testAccRolePermissionOneAssigned, strings.ToLower(t.Name())),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.auth0_role.role", "permissions.#", "1"),
+					resource.TestCheckResourceAttr("data.auth0_role.role", "permissions.0.name", "read:foo"),
+					resource.TestCheckResourceAttr("data.auth0_role.role", "permissions.0.resource_server_identifier", "https://uat.testaccrolepermissions.terraform-provider-auth0.com/api"),
+
+					resource.TestCheckResourceAttr("auth0_role_permissions.role_permissions", "permissions.0.name", "read:foo"),
+					resource.TestCheckResourceAttr("auth0_role_permissions.role_permissions", "permissions.0.resource_server_identifier", "https://uat.testaccrolepermissions.terraform-provider-auth0.com/api"),
+					resource.TestCheckResourceAttr("auth0_role_permissions.role_permissions", "permissions.0.resource_server_name", "Acceptance Test - testaccrolepermissions"),
+					resource.TestCheckResourceAttr("auth0_role_permissions.role_permissions", "permissions.0.description", "Can read Foo"),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(testAccRolePermissionTwoAssigned, strings.ToLower(t.Name())),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("auth0_role_permissions.role_permissions", "permissions.0.name", "create:foo"),
+					resource.TestCheckResourceAttr("auth0_role_permissions.role_permissions", "permissions.1.name", "read:foo"),
+
+					resource.TestCheckResourceAttr("data.auth0_role.role", "permissions.#", "2"),
+					resource.TestCheckResourceAttr("data.auth0_role.role", "permissions.0.name", "create:foo"),
+					resource.TestCheckResourceAttr("data.auth0_role.role", "permissions.1.name", "read:foo"),
+
+					resource.TestCheckResourceAttr("data.auth0_role.role", "permissions.0.resource_server_identifier", "https://uat.testaccrolepermissions.terraform-provider-auth0.com/api"),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(testAccRolePermissionOneAssigned, strings.ToLower(t.Name())),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.auth0_role.role", "permissions.#", "1"),
+					resource.TestCheckResourceAttr("data.auth0_role.role", "permissions.0.name", "read:foo"),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(testAccRolePermissionNoneAssigned, strings.ToLower(t.Name())),
+			},
+			{
+				RefreshState: true,
```

Comment:
Impostor!

Reply:
This is necessary otherwise the data source doesn't get the change reflected.

Reply:
Ah, I see that now 👍🏻 , indeed the execution flow on a terraform apply is:

0. read and refresh state
1. creations
2. updates
3. deletions

so we need another refresh state after the deletion.